include/zephyr/toolchain/common.h

```diff
@@ -274,6 +274,17 @@
  * @}
  */ /* end of struct_section_apis */
 
+/** @brief Tag an object (e.g. function) to be kept in the binary even though it is not used.
+ *
+ * It prevents object from being removed by the linker garbage collector. It
+ * is achieved by adding a pointer to that object to the kept memory section.
+ *
+ * @param object Function or variable name.
+ */
+#define KEEP_OBJECT(object) \
```

Comment:
More nitpickery: maybe "LINKER_KEEP()" or something would be better, to clarify that the macro does the same thing that the KEEP() expression does in a linker script?  The word "object" is ambiguous basically everywhere in software engineering.  Using "symbol" would be better if you want a noun IMHO, as that's the name of the the abstraction being managed.

Also, you can relax the documentation a little, the argument can be any pointer expression that points at compile time into linker-visible memory. `KEEP_OBJECT(global_object.field[7])` would work just fine to save the global_object.  (This is silly as written, but it's easy to imagine the argument itself being a macro that expands to some selected subobject or whatever).

Reply:
Renamed to `LINKER_KEEP` and use `symbol` term instead of object.